### PR TITLE
Limit premium domain purchases that exceed the price limit

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* We need to limit certain premium domain price points, so we'll get an extra property in the domain pricing API response. We should also not allow premium domains over that price point to be added to the cart.

Here's how it looks:

**when loading the price:**
Screenshot 2020-09-22 at 13 36 33

**when the price is loaded and it's above the price limit:**
Screenshot 2020-09-23 at 10 21 19

**what you see if you click on the info badge:**
Screenshot 2020-09-23 at 10 22 31

see: p99Zz8-1qh-p2

#### Testing instructions

* in the dev or wpcalypso env search for a premium domain that costs above 6000 (search for image.blog for example). Make sure that once the price is loaded the badge changes to "Restricted premium". Verify that the info popover text also changes.

### CODE REVIEW

**Comment by hambai** [comment]
Hey @fditrapani - sorry to bother you but can you share your opinion on the "Restricted premium" badge?

File: client/components/domains/premium-badge/index.jsx [hunk 1]
```diff
@@ -13,16 +14,34 @@ import InfoPopover from 'components/info-popover';
...
 
 class PremiumBadge extends React.Component {
-	render() {
+	getPopoverText( restrictedPremium ) {
 		const { translate } = this.props;
+		if ( restrictedPremium ) {
+			return {
+				text: translate( 'Restricted premium' ),
+				description: translate(
+					"This premium domain tier is currently not available at WordPress.com. Please contact our support if you're interested."
```
Comment (daledupreez) [inline client/components/domains/premium-badge/index.jsx:23]: I'd prefer to update the language here to:
> This premium domain is currently not available at WordPress.com. Please contact support if you're interested in this domain.

**Review by daledupreez — APPROVED** [review]
Functionally, this works as described, but I'd also like Filippo to take a quick look. One minor concern I have is that the contrast between the white text and the orange background might be a bit low.

I think it's also worth noting that D49943-code is needed on your back end to fully test this.

File: client/components/domains/premium-badge/index.jsx (hunk 1 above)
Comment (daledupreez) [reply client/components/domains/premium-badge/index.jsx]: I fixed this by inverting the colour for the text and the icon.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4504611

Thank you @hambai for including a screenshot in the description! This is really helpful for our translators.

**Comment by fditrapani** [comment]
Sorry for the late reply — I was AFK last week. I checked the contrast on the text colour and background and it passes our accessibility requirements. Thanks for the ping!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4504611).

**Pull request merged**